hazelcast/src/main/java/com/hazelcast/spi/PartitionMigrationEvent.java

```diff
@@ -54,8 +61,33 @@ public int getPartitionId() {
         return partitionId;
     }
 
+    /**
+     * Gets the index of the partition replica that current member owns currently, before migration starts.
+     * This index will be in range of [0,6] if current member owns a replica of the partition. Otherwise it will be -1.
+     *
+     * @return index of the partition replica that current member owns currently
+     */
+    public int getCurrentReplicaIndex() {
+        return currentReplicaIndex;
+    }
+
+    /**
+     * Gets the index of the partition replica that current member will own after migration is committed.
+     * This index will be -1 if partition replica will be moved from current member completely.
+     *
+     * @return index of the partition replica that current member will own after migration is committed
+     */
+    public int getNewReplicaIndex() {
```

Comment:
do you think some sort of guidance help to implementers? 
by looking at our current implementations the logic seem to be quite similar.